src/app-logic/url-handling.js

```diff
@@ -135,19 +144,33 @@ type JsTracerQuery = {|
   summary: null | void,
 |};
 
-// Use object type spread in the definition of Query rather than unions, so that they
-// are really easy to manipulate. This permissive definition makes it easy to not have
-// to refine the type down to the individual query types when working with them.
-type Query = {|
-  ...CallTreeQuery,
-  ...MarkersQuery,
-  ...NetworkQuery,
-  ...StackChartQuery,
-  ...JsTracerQuery,
-|};
+type Query =
+  | CallTreeQuery
+  | MarkersQuery
+  | NetworkQuery
+  | StackChartQuery
+  | JsTracerQuery;
 
 type $MakeOptional = <T>(T) => T | void;
-type QueryShape = $Shape<$ObjMap<Query, $MakeOptional>>;
+type BaseQueryShape = $Shape<$ObjMap<BaseQuery, $MakeOptional>>;
```

Comment:
I think it would be good to provide a bit of documentation to these different types on why they are needed. There are enough here that it's a little intimidating to look at and now how to work with.